src/main/java/dev/blachut/svelte/lang/Svelte.flex

```diff
@@ -103,6 +103,8 @@ WHITE_SPACE=\s+
 <TAG_STRING> {
   "'"                         { if (quote == '\'') yybegin(HTML_TAG); return HTML_FRAGMENT; }
   "\""                        { if (quote == '"') yybegin(HTML_TAG); return HTML_FRAGMENT; }
+  "{"                         { yybegin(SVELTE_ATTR); return START_MUSTACHE; }
+  "}"                         { if (leftBraceCount == 0) { yybegin(YYINITIAL); return END_MUSTACHE; } else { leftBraceCount -= 1; return CODE_FRAGMENT; } }
```

Comment:
@unlocomqx - not sure if "}" case is needed here, as we never return to "TAG_STRING" after "{", because it will be handled with SVELTE_ATTR.

Reply:
I moved that code from there to a seprate `SVELTE_ATTR` block but been looking for hours on how to return to the previous state, I will try `yybegin(TAG_HTML)` for now to see what it does

Reply:
I don't think that always return to TAG_HTML is a good idea, because you can be inside a string: `<SomeTag value='{blah}' />` When you detect "{", then you need to return to TAG_STRING. But situation is more complex, as you could have another inner string or curly brackets: `<SomeTag value='{(function() { return "blah"})()}' />`. Here we have 2 levels for strings and 2 levels for brackets. I am thinking about state stack like this: https://stackoverflow.com/questions/887485/keeping-track-of-state-in-jflex. What do you think?

Reply:
The `leftBraceCount` would take care of nesting but using the stack to return to last state is a great idea, I will give it a shot

Reply:
Double thinking about this issue, I don't think that we actually need a state stack. The only we need is to return to HTML_TAG or TAG_STRING from the SVELTE_TAG depending on the previous state, so a single flag for this is enough. And we don't need to care about inner strings or curly brackets. They will be part of the outer CODE_FRAGMENT. `leftBraceCount` plays a good role here.

I also think that we need to extract SVELTE_TAG from the `<SVELTE_INTERPOLATION, SVELTE_TAG, SVELTE_ATTR, SVELTE_TAG_PAREN_AWARE>` (for you this line might be different a little bit), because we need to return to `YYINITIAL` in case of SVELTE_TAG/"}", and to HTML_TAG or TAG_STRING from the SVELTE_TAG.

Reply:
I used the variable `previousState` to track the state
![screenshot_24](https://user-images.githubusercontent.com/793712/61476537-2baccc00-a985-11e9-954c-35d72c8585bb.png)
It works better now but the second attribute isn't correctly highlighted

Reply:
I made similar approach as your: https://github.com/tiatin/svelte-intellij/commit/82d78768cd1469beb79a9d5cb77802b8e1dfc4d3. For me it works fine with multiple attributes. Also created a new release: https://github.com/tiatin/svelte-intellij/releases/tag/v0.7.3

I see you try to reuse SVELTE_INTERPOLATION, but I think it might fail in case or real SVELTE_INTERPOLATION, because in this case we need to always return to YYINITIAL, but previousState is set initially to 0: I am not sure if YYINITIAL will be 0 in the DFA.

Reply:
Regarding the attribute, I mean't that the attribute name is not highlighted correctly
The nesting seems to break the html highlighting (notice the z attribute)
![screenshot_25](https://user-images.githubusercontent.com/793712/61529247-cefaf100-aa18-11e9-9245-c73647c90524.png)